src/Surfnet/AzureMfa/Infrastructure/Controller/DefaultController.php

```diff
@@ -106,6 +106,13 @@ public function registrationAction(Request $request)
             return $this->registrationService->replyToServiceProvider();
         }
 
+        $attribs = $this->authenticationService->getGsspUserAttributes();
+        if ($attribs && $attribs->getAttributeValue('urn:mace:dir:attribute-def:mail')) {
+            $emailAddr = new EmailAddress($attribs->getAttributeValue('urn:mace:dir:attribute-def:mail'));
+            $user = $this->azureMfaService->startRegistration($emailAddr);
+            return new RedirectResponse($this->azureMfaService->createAuthnRequest($user));
+        }
```

Comment:
Nice and simple. For the sake of single responsibility I'd prefer seeing this logic moved out to a service. But as the rest of the controller is already chock-full of business logic. This kind of ties the room together.